autocorpus/word_extractor.py

```diff
@@ -0,0 +1,165 @@
+"""This module provides functionality to extract text and tables from Word documents (.doc and .docx).
+
+It includes methods to handle older .doc files by converting them to .docx format and processing them.
+"""
+
+import os
+import platform
+import subprocess
+from pathlib import Path
+
+from docx import Document
+
+from autocorpus.ac_bioc.bioctable.collection import BioCTableCollection
+from autocorpus.ac_bioc.bioctable.json import BioCTableJSON
+from autocorpus.ac_bioc.collection import BioCCollection
+from autocorpus.ac_bioc.json import BioCJSON
+from autocorpus.bioc_supplementary import BioCTableConverter, BioCTextConverter
+
+from . import logger
+
+
+def __extract_tables(doc):
+    """Extracts tables from a .docx document.
+
+    Args:
+        doc (docx.Document): The Document object representing the .docx document.
+
+    Returns:
+        list: A list of tables extracted from the document. Each table is represented as a nested list,
+              where each inner list corresponds to a row, and each element in the inner list corresponds
+              to the text content of a cell in the row.
+
+    Example:
+        from docx import Document
+
+        doc = Document("document.docx")
+        tables = extract_tables(doc)
+    """
+    # Open the .docx file
+    tables = []
+    # Iterate through the tables in the document
+    for table in doc.tables:
+        tables.append([])
+        # Iterate through the rows in the table
+        for row in table.rows:
+            tables[-1].append([x.text for x in row.cells])
+    return tables
+
+
+def __convert_older_doc_file(file: Path, output_dir: Path) -> Path | bool:
+    """Converts an older .doc file to .docx format using platform-specific methods."""
+    operating_system = platform.system()
+    docx_path = Path(str(file).replace(".doc", ".docx"))
+    if operating_system == "Windows":
+        import win32com.client
+
+        word = None
+        try:
+            word = win32com.client.DispatchEx("Word.Application")
+            doc = word.Documents.Open(file)
+            doc.SaveAs(docx_path, 16)
+            doc.Close()
+            word.Quit()
+            return docx_path
+        except Exception:
+            return False
+        finally:
+            if word:
+                word.Quit()
+    elif operating_system == "Linux":
```

Comment:
This `if`/`elif` chain doesn't have a fallback `else`. I'd fix that for code hygiene reasons, but I think you can just try the LibreOffice approach for the non-Windows, non-macOS case because it should also work for the (admittedly unlikely) case that someone's running FreeBSD or whatever